Pellicola/Classes/PellicolaFileHandler.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  PellicolaFileHandler.swift
+//  Pellicola
+//
+//  Created by marco.rossi on 18/10/2019.
+//
+
+import Foundation
+
+@objcMembers
+public final class PellicolaFileHandler: NSObject {
+    private let fileManager: FileManager
+    
+    public static func defaultHandler() -> PellicolaFileHandler {
+        return PellicolaFileHandler()
+    }
+    
+    public init(fileManager: FileManager = FileManager.default) {
+        self.fileManager = fileManager
+    }
+    
+    public func saveImage(_ image: UIImage, named: String) -> URL? {
+        guard let url = fileURL(forIdentifier: named), let imageData = image.pngData() else { return nil }
+        do {
+            try createSubfoldersBeforeCreatingFile(at: url)
+            try imageData.write(to: url, options: .atomic)
+            return url
+        } catch {
+            return nil
+        }
+    }
+    
+    public func deleteImage(at url: URL) {
+        try? fileManager.removeItem(at: url)
+    }
+    
+    public func deleteAllImages() {
+        guard let cacheFolder = self.cacheFolder() else { return }
+        try? fileManager.removeItem(at: cacheFolder)
+    }
+    
+    private func fileURL(forIdentifier identifier: String) -> URL? {
+        return cacheFolder()?.appendingPathComponent(identifier + ".png")
+    }
+    
+    private func cacheFolder() -> URL? {
+        guard let cacheDirURL = fileManager.urls(for: .cachesDirectory, in: .userDomainMask).first else { return nil }
+        return cacheDirURL.appendingPathComponent("Pellicola_Images")
+    }
+    
+    private func createSubfoldersBeforeCreatingFile(at url: URL) throws {
+        do {
+            let subfolderUrl = url.deletingLastPathComponent()
+            var subfolderExists = false
+            var isDirectory: ObjCBool = false
+            if fileManager.fileExists(atPath: subfolderUrl.path, isDirectory: &isDirectory) {
```

Comment:
If I am not mistaken `createDirectory(at: ..., withIntermediateDirectories: true)` will succeed when the specified directory already exists. You can probably invoke it directly without checking whether the folder already exists.